### PR TITLE
Fix reader action handler update from database

### PR DESCRIPTION
**Pull request by mvasilak**
* Uses correct indexes if there are invalid annotations
* Improves free text annotation validation

### CODE REVIEW

File: Zotero/Scenes/Detail/PDF/ViewModels/PDFReaderActionHandler.swift [hunk 1]
```diff
@@ -2135,8 +2136,17 @@ final class PDFReaderActionHandler: ViewModelActionHandler, BackgroundDbProcessi
 
         DDLogInfo("PDFReaderActionHandler: database annotation changed")
 
-        // Get sorted database keys
-        var keys = (viewModel.state.snapshotKeys ?? viewModel.state.sortedKeys).filter({ $0.type == .database })
+        // Keys for deletions and modifications are indexed by the previously observed items
+        var previousObservedKeys: [PDFReaderState.AnnotationKey] = []
```
Comment (michalrentka) [inline Zotero/Scenes/Detail/PDF/ViewModels/PDFReaderActionHandler.swift:2140]: Can we just directly use `viewModel.state.databaseAnnotations`? We do that on line 2225 anyway. No real need to extract keys separately.

File: Zotero/Scenes/Detail/PDF/ViewModels/PDFReaderActionHandler.swift [hunk 2]
```diff
@@ -2135,8 +2136,17 @@ final class PDFReaderActionHandler: ViewModelActionHandler, BackgroundDbProcessi
...
+        for item in viewModel.state.databaseAnnotations {
+            previousObservedKeys.append(.init(key: item.key, type: .database))
+        }
+        // Keys for insertions are indexed by the currently observed items
+        var currentObservedKeys: [PDFReaderState.AnnotationKey] = []
```
Comment (michalrentka) [inline Zotero/Scenes/Detail/PDF/ViewModels/PDFReaderActionHandler.swift:2145]: Does this even need to be created? I only see `currentObservedKeys.count` being used, so you could just do `objects.count` or store `let currentKeysCount = objects.count` and use that for clarity.

**Review by michalrentka — CHANGES_REQUESTED** [review]

File: Zotero/Scenes/Detail/PDF/ViewModels/PDFReaderActionHandler.swift (hunk 1 above)
Comment (mvasilak) [reply Zotero/Scenes/Detail/PDF/ViewModels/PDFReaderActionHandler.swift]: Good point (I planned to improve that, after changing font size to use a float, but better if I do it in a separate PR)
Check https://github.com/zotero/zotero-ios/pull/999/commits/6e0b28fbb1a3b464b94c6cc4bbe698c5c366c8cb

File: Zotero/Scenes/Detail/PDF/ViewModels/PDFReaderActionHandler.swift (hunk 2 above)
Comment (mvasilak) [reply Zotero/Scenes/Detail/PDF/ViewModels/PDFReaderActionHandler.swift]: Check https://github.com/zotero/zotero-ios/pull/999/commits/6e0b28fbb1a3b464b94c6cc4bbe698c5c366c8cb

**Review by michalrentka — APPROVED** [review]

**Pull request merged**